llvm/lib/Target/DirectX/DXILShaderFlags.cpp

```diff
@@ -63,20 +106,64 @@ void ComputedShaderFlags::print(raw_ostream &OS) const {
   OS << ";\n";
 }
 
+/// Get the combined shader flag mask of all module functions.
+const ComputedShaderFlags DXILModuleShaderFlagsInfo::getCombinedFlags() const {
+  return CombinedSFMask;
+}
+
+/// Return the shader flags mask of the specified function Func, if one exists.
+/// else an error
+Expected<const ComputedShaderFlags &>
+DXILModuleShaderFlagsInfo::getShaderFlagsMask(const Function *Func) const {
+  std::pair<Function const *, ComputedShaderFlags> V{Func, {}};
+  const auto Iter = llvm::lower_bound(FunctionFlags, V);
```

Comment:
Probably simpler to use a comparator function that just compares against the pair's `first` rather than constructing a sentinel to compare against.

Also llvm style prefers `const Function *` to `Function const *`

Reply:
Changed to use comparator.